compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SymbolResolver.java

```diff
@@ -1947,24 +1947,44 @@ public BSymbol getArithmeticOpsForTypeSets(OperatorKind opKind, BType lhsType, B
                 compatibleType2 = types.findCompatibleType(rhsType);
             }
 
-            if (types.isBasicNumericType(compatibleType1) && compatibleType1 != compatibleType2) {
+            if (types.isBasicNumericType(compatibleType1) && compatibleType1 != compatibleType2 && 
+                    !isIntFloatingPointMultiplication(opKind, compatibleType1, compatibleType2)) {
                 return symTable.notFoundSymbol;
             }
+
+            BType returnType;
             if (compatibleType1.tag < compatibleType2.tag) {
-                return createBinaryOperator(opKind, lhsType, rhsType, compatibleType2);
-            }
-            if (lhsType.isNullable()) {
-                compatibleType1 = BUnionType.create(null, compatibleType1, symTable.nilType);
-                return createBinaryOperator(opKind, lhsType, rhsType, compatibleType1);
+                returnType = compatibleType2;
+            } else {
+                returnType = compatibleType1;
```

Comment:
Shall we inline this as below,
`BType returnType = compatibleType1.tag < compatibleType2.tag ? compatibleType1 : compatibleType2;`